config/pmd/ruleset.xml

```diff
@@ -1,75 +1,107 @@
-    <exclude name="TooManyStaticImports" />
-  </rule>
-  <rule ref="category/java/codestyle.xml/ClassNamingConventions">
-    <properties>
-      <!-- same as any other class -->
-      <property name="utilityClassPattern" value="[A-Z][a-zA-Z]+" />
-    </properties>
-  </rule>
-  <rule ref="category/java/codestyle.xml/MethodNamingConventions">
-    <properties>
-      <property name="junit4TestPattern" value="[a-z][a-zA-Z0-9_]+" />
-    </properties>
-  </rule>
-  <rule ref="category/java/codestyle.xml/ShortVariable">
-    <properties>
-      <property name="minimum" value="2" />
-    </properties>
-  </rule>
-  <rule ref="category/java/codestyle.xml/VariableNamingConventions">
-    <properties>
-      <property name="violationSuppressRegex" value=".*'log'.*" />
-    </properties>
-  </rule>
-  <rule ref="category/java/design.xml">
-    <exclude name="AvoidCatchingGenericException" />
-    <exclude name="UseUtilityClass" />
-    <exclude name="LoosePackageCoupling" />
-    <exclude name="DataClass" />
-  </rule>
-  <rule ref="category/java/design.xml/ExcessiveImports">
-    <properties>
-      <property name="minimum" value="35" /><!-- should be reduced -->
-    </properties>
-  </rule>
-  <rule ref="category/java/design.xml/LawOfDemeter">
-    <properties>
-      <property name="violationSuppressRegex" value="(.*method chain calls.*|.*object not created locally.*)" />
-    </properties>
-  </rule>
-  <rule ref="category/java/design.xml/SignatureDeclareThrowsException">
-    <properties>
-      <property name="IgnoreJUnitCompletely" value="true" />
-    </properties>
-  </rule>
-  <rule ref="category/java/documentation.xml">
-    <exclude name="CommentRequired" />
-    <exclude name="CommentSize" />
-    <exclude name="UncommentedEmptyMethodBody" />
-  </rule>
-  <rule ref="category/java/errorprone.xml">
-    <exclude name="BeanMembersShouldSerialize" />
-  </rule>
-  <rule ref="category/java/multithreading.xml" />
-  <rule ref="category/java/performance.xml" />
-  <rule ref="category/java/security.xml" />
+    <rule ref="category/java/bestpractices.xml">
+        <exclude name="GuardLogStatement"/>
+        <exclude name="JUnitTestContainsTooManyAsserts"/>
+    </rule>
+    <rule ref="category/java/bestpractices.xml/JUnitTestsShouldIncludeAssert">
+        <properties>
+            <property name="violationSuppressXPath" value="//ClassOrInterfaceType[@Image='RestAssuredTest']"/>
+        </properties>
+    </rule>
+    <rule name="RestAssuredTestHasAssertions"
+          language="java"
+          message="a test should have assertions - using either assert or then"
+          class="net.sourceforge.pmd.lang.rule.XPathRule">
+        <description>
+            RestAssured tests should have assertions either using 'assert...' or 'then'
+        </description>
+        <priority>3</priority>
+        <properties>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+//ClassOrInterfaceType[
```

Comment:
is it needed? or can we format it as least?

Reply:
Yes - this is needed. That's how the custom rule is defined